packages/simpleauth/src/TeamsFxSimpleAuth.Tests/Helpers/Utilities.cs

```diff
@@ -101,18 +102,25 @@ private static void ClickElementWithRetry(IWebDriver driver, By elementLocator,
 
             // Click "Accept"
             ClickElementWithRetry(driver, By.Id("idSIButton9"));
-
+            
             Uri redirectedUrl = new Uri(driver.Url);
-            driver.Close();
-
             string authorizationCode = HttpUtility.ParseQueryString(redirectedUrl.Query).Get("code");
+            int retryTime = 0;
 
-            if (string.IsNullOrEmpty(authorizationCode))
+            do
             {
-                throw new Exception($"Failed to get authorization code from redirect url: {redirectedUrl.AbsoluteUri}");
-            }
+                if(!string.IsNullOrEmpty(authorizationCode))
+                {
+                    driver.Close();
+                    return authorizationCode;
+                }
+                Thread.Sleep(200 * ++retryTime);
```

Comment:
How about using await Task.delay(200) instead of Thread.Sleep? Since Thread.Sleep is blocking

Reply:
this is a static method, not a task, so I think thread.sleep is better.